src/pkg/extensions/extensions_test.go

```diff
@@ -0,0 +1,184 @@
+				NewMockExtension,
+			},
+			rc: testRc,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err == nil && extRc != nil && extInfo != nil
+			},
+		},
+		{
+			name: "multiple valid factories",
+			factories: []CorsoItemExtensionFactory{
+				NewMockExtension,
+				NewMockExtension,
+				NewMockExtension,
+			},
+			rc: testRc,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err == nil && extRc != nil && extInfo != nil
+			},
+		},
+		{
+			name:      "no factories supplied",
+			factories: nil,
+			rc:        testRc,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err != nil && extRc == nil && extInfo == nil
+			},
+		},
+		{
+			name: "factory slice contains nil",
+			factories: []CorsoItemExtensionFactory{
+				NewMockExtension,
+				nil,
+				NewMockExtension,
+			},
+			rc: testRc,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err != nil && extRc == nil && extInfo == nil
+			},
+		},
+		{
+			name: "factory call returns error",
+			factories: []CorsoItemExtensionFactory{
+				func(
+					ctx context.Context,
+					rc io.ReadCloser,
+					info details.ItemInfo,
+					extInfo *details.ExtensionInfo,
+				) (CorsoItemExtension, error) {
+					return nil, clues.New("creating extension")
+				},
+			},
+			rc: testRc,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err != nil && extRc == nil && extInfo == nil
+			},
+		},
+		{
+			name: "one or more factory calls return error",
+			factories: []CorsoItemExtensionFactory{
+				NewMockExtension,
+				func(
+					ctx context.Context,
+					rc io.ReadCloser,
+					info details.ItemInfo,
+					extInfo *details.ExtensionInfo,
+				) (CorsoItemExtension, error) {
+					return nil, clues.New("creating extension")
+				},
+			},
+			rc: testRc,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err != nil && extRc == nil && extInfo == nil
+			},
+		},
+		{
+			name: "nil inner rc",
+			factories: []CorsoItemExtensionFactory{
+				NewMockExtension,
+			},
+			rc: nil,
+			validateOutputs: func(
+				extRc io.ReadCloser,
+				extInfo *details.ExtensionInfo,
+				err error,
+			) bool {
+				return err != nil && extRc == nil && extInfo == nil
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+			ith := &ItemExtensionHandler{}
+
+			extRc, extInfo, err := ith.AddItemExtensions(
+				ctx,
+				test.rc,
+				testItemInfo,
+				test.factories)
+			require.True(t, test.validateOutputs(extRc, extInfo, err))
```

Comment:
For full coverage, this test should read and close the reader on each run (which means you'll also need to construct a new base reader).  That will allow you to:
1. ensure the item is read properly.
2. exercise tests for different read chunks and ensure that works properly.
3. exercise the close func.

Reply:
yes! This is covered in a later PR. `mock_extensions.go` has some code which lays the groundwork for read/close testing.
I've only added tests here for  `AddItemExtensions` which only initializes the extensions.

Thanks for raising it though! Really appreciate it!